bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/link/ItemChannelLinkResource.java

```diff
@@ -80,9 +82,13 @@ public Response isAutomatic() {
     @Path("/{itemName}/{channelUID}")
     @ApiOperation(value = "Links item to a channel.")
     @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
-            @ApiResponse(code = 400, message = "Item already linked to the channel.") })
+            @ApiResponse(code = 400, message = "Item already linked to the channel."),
+            @ApiResponse(code = 405, message = "Item name invalid.") })
     public Response link(@PathParam("itemName") @ApiParam(value = "itemName") String itemName,
             @PathParam("channelUID") @ApiParam(value = "channelUID") String channelUid) {
+        if (!itemRegistry.isValidItemName(itemName)) {
```

Comment:
But perhaps there is some reason, that the service interface provides that API and it is up to a special implementation.
WDYT?
Currently the "is valid" is implementation specific for the used item registry. If we use a static method this is not the case anymore.